# --- T2-COPYRIGHT-BEGIN ---
# t2/package/*/osmo/hotfix-c23.patch
# Copyright (C) 2026 The T2 SDE Project
# SPDX-License-Identifier: GPL-2.0 or patched project license
# --- T2-COPYRIGHT-END ---

Add full prototype for the note callback, C23 requires it.

Copyright (C) 2026 René Rebe <rene@exactco.de>
Signed-off-by: René Rebe <rene@exactco.de>

--- a/src/calendar_notes.h	2017-03-28 22:27:38.000000000 +0200
+++ b/src/calendar_notes.h	2026-09-25 19:43:34.159627508 +0200
@@ -53,7 +53,7 @@ enum {
 
 gboolean    cal_check_note              (guint32 julian, GUI *appGUI);
 gboolean    cal_check_notes             (guint32 julian_start, guint32 julian_end, GUI *appGUI);
-void        cal_notes_foreach           (guint32 julian_start, guint32 julian_end, gboolean (*cnfunc)(), GUI *appGUI);
+void        cal_notes_foreach           (guint32 julian_start, guint32 julian_end, gboolean (*cnfunc)(struct note *, GUI *), GUI *appGUI);
 gchar *     cal_get_note                (guint32 julian, GUI *appGUI);
 gchar *     cal_get_note_color          (guint32 julian, GUI *appGUI);
 void        cal_replace_note_color      (gchar *old_color, gchar *new_color, GUI *appGUI);
--- a/src/calendar_notes.c	2020-06-14 08:04:47.000000000 +0200
+++ b/src/calendar_notes.c	2026-09-25 19:43:34.158978185 +0200
@@ -67,7 +67,7 @@ cal_check_notes (guint32 julian_start, g
 /*------------------------------------------------------------------------------*/
 
 void
-cal_notes_foreach (guint32 julian_start, guint32 julian_end, gboolean (*cnfunc)(), GUI *appGUI)
+cal_notes_foreach (guint32 julian_start, guint32 julian_end, gboolean (*cnfunc)(struct note *, GUI *), GUI *appGUI)
 {
 	GSList *node;
 	struct note *n;
